### PR TITLE
Using robust Huber loss in Translation Averaging

### PR DESCRIPTION
**Pull request by akshay-krishnan**
The metrics on skydio-32 show a clear improvement: 

Before the change:

|Metric | Value|
|---|---|
|mean_translation_angle_error_deg | 16.2499|
|mean_translation_error_distance | 0.98776 |

After the change:

|Metric | Value|
|---|---|
|mean_translation_angle_error_deg | 10.3671|
|mean_translation_error_distance | 0.546729 |

There are also small improvements after bundle adjustment. The improvement is similar on skydio-8, not too significant on door. 
I see from the plots that the inlier errors significantly decrease, but large outliers are not quite affected, which is expected.  

[CI run for plots](https://github.com/borglab/gtsfm/actions/runs/1223467854)

### CODE REVIEW

**Comment by johnwlambert** [comment]
Looks awesome @akshay-krishnan. Can we make it a default flag set to true to use_robust in the translation averaging? That way in ablation tables we can just flip the switch

**Comment by akshay-krishnan** [comment]
> Looks awesome @akshay-krishnan. Can we make it a default flag in the config set to true to use_robust in the translation averaging? That way in ablation tables we can just flip the switch

Yeah, but we would have change the base class API right? And this is something implementation-specific. What if there is a translation averaging implementation which does not need that flag?

**Comment by johnwlambert** [comment]
> > Looks awesome @akshay-krishnan. Can we make it a default flag in the config set to true to use_robust in the translation averaging? That way in ablation tables we can just flip the switch
> 
> Yeah, but we would have change the base class API right? And this is something implementation-specific. What if there is a translation averaging implementation which does not need that flag?
> Do we have any implementation-specific flags in other classes?

That's true. But I still think it's a good idea. Could be an optional flag that defaults to False.
I like how Ayush did it here: https://github.com/borglab/gtsfm/pull/70/files

Maybe for some implementations, making it robust is tricky, so we would keep default false.

**Comment by akshay-krishnan** [comment]
> > > Looks awesome @akshay-krishnan. Can we make it a default flag in the config set to true to use_robust in the translation averaging? That way in ablation tables we can just flip the switch
> > 
> > 
> > Yeah, but we would have change the base class API right? And this is something implementation-specific. What if there is a translation averaging implementation which does not need that flag?
> > Do we have any implementation-specific flags in other classes?
> 
> That's true. But I still think it's a good idea. Could be an optional flag that defaults to False.
> I like how Ayush did it here: https://github.com/borglab/gtsfm/pull/70/files
> 
> Maybe for some implementations, making it robust is tricky, so we would keep default false.

Not exactly the same case, since there is no bundle adjustment base class. But sure, I can add a flag. 
What I mean is, the flag might be meaningless for some other implementation, it's a base class argument that is implementation-specific, which is not very clean API-wise.

**Comment by akshay-krishnan** [comment]
Done, and that will also be my first commit using github.dev :)

File: gtsfm/averaging/translation/averaging_1dsfm.py [hunk 1]
```diff
@@ -64,7 +67,12 @@ def run(
                 or ill-constrained system).
             A GtsfmMetricsGroup of 1DSfM metrics.
         """
-        noise_model = gtsam.noiseModel.Isotropic.Sigma(NOISE_MODEL_DIMENSION, NOISE_MODEL_SIGMA)
+        isotropic_noise_model = gtsam.noiseModel.Isotropic.Sigma(NOISE_MODEL_DIMENSION, NOISE_MODEL_SIGMA)
```
Comment (ayushbaid) [inline gtsfm/averaging/translation/averaging_1dsfm.py:70]: just rename the first variable as noise_model, and overwrite it if the flag is true?

**Review by ayushbaid — APPROVED** [review]
LGTM except one minor comment.

**Comment by ayushbaid** [comment]
Redacting approval because John's comment needs to be addressed.

**Comment by akshay-krishnan** [comment]
> should this line be
> 
> ```python
> noise_model = gtsam.noiseModel.Robust.Create(huber_loss, isotropic_noise_model)
> ```
> 
> ?

Good catch, too bad my first github.dev commit was buggy :/

**Comment by johnwlambert** [comment]
Looks really good Akshay. Can we make it an argument of the constructor instead though? And not extra arg to run? That way we can flip switch in config. Can we put it True in the config too?

**Review by dellaert — APPROVED** [review]
This is awesome!

**Review by johnwlambert — APPROVED** [review]
Nice work, thanks Akshay

**Pull request merged**